models/gc_spider_baseline/utils/SpiderBaselineRunner.py

```diff
@@ -0,0 +1,155 @@
+
+
+@IO.Config('traversal_direction_up', bool, True, the="Direction to traverse the image")
+@IO.ConfigInput('in_data', 'nifti|nrrd|mha:mod=ct|mr', the='supported datatypes for the spider baseline model')
+class SpiderBaselineRunner(Module):
+
+    traversal_direction_up: bool
+
+    SPIDER_DATA_DIR = Path(os.environ["VERSEG_BASEDIR"])
+    SPIDER_INTERNAL_DATASET_PATH = SPIDER_DATA_DIR / "datasets" / "spider_input"
+    SPIDER_EXPERIMENT_DIR = SPIDER_DATA_DIR / "experiments" / "SPIDER-Baseline"
+    SPIDER_INTERNAL_OUTPUT_DIR = SPIDER_EXPERIMENT_DIR / "results" / "spider_input"
+    SPIDER_EXPERIMENT_ARGUMENTS_FILE = SPIDER_EXPERIMENT_DIR / "arguments.json"
+    SPIDER_INTERNAL_DATASET_META_FILE = SPIDER_INTERNAL_DATASET_PATH / "metadata.json"
+
+    @IO.Instance()
+    @IO.Input('in_data', the='input sagittal spine image')
+    @IO.Output(
+        'out_data_raw',
+        'spider_baseline_vertebrae_segmentation_raw.mha',
+        'mha:mod=seg:model=SpiderBaselineSegmentation:seg=raw',
+        data='in_data',
+        bundle='model',
+        the='Spider baseline vertebrae segmentation for the input sagittal spine MRI. Raw segmentation output: '
+            '0: background '
+            '1-25: different vertebrae numbered from the bottom (i.e. L5 = 1) '
+            '100: spinal canal '
+            '101-125 for partially visible vertebrae '
+            '201-225: different intervertebral discs (also numbered from the bottom, i.e. L5/S1 = 201)) '
+    )
+    @IO.Output(
+        'out_data',
+        'spider_baseline_vertebrae_segmentation.mha',
+        'mha:mod=seg:model=SpiderBaselineSegmentation:seg=remapped:roi=VERTEBRAE_L5,VERTEBRAE_L4,VERTEBRAE_L3,VERTEBRAE_L2,VERTEBRAE_L1,VERTEBRAE_T12,VERTEBRAE_T11,VERTEBRAE_T10,VERTEBRAE_T9,VERTEBRAE_T8,VERTEBRAE_T7,VERTEBRAE_T6,VERTEBRAE_T5,VERTEBRAE_T4,VERTEBRAE_T3,VERTEBRAE_T2,VERTEBRAE_T1,VERTEBRAE_C7,VERTEBRAE_C6,VERTEBRAE_C5,VERTEBRAE_C4,VERTEBRAE_C3,VERTEBRAE_C2,VERTEBRAE_C1',
+        data='in_data',
+        bundle='model',
+        the='Spider baseline vertebrae segmentation for the input sagittal spine MRI. Remapped segmentation output: '
+            '0: background '
+            '1-25: different vertebrae numbered from the bottom (i.e. L5 = 1) (includes partially visible vertebrae from raw segmentation)'
+    )
+    def task(self, instance: Instance, in_data: InstanceData, out_data_raw: InstanceData, out_data: InstanceData) -> None:
+        in_data_path = Path(in_data.abspath)
+        internal_img_path = self.SPIDER_INTERNAL_DATASET_PATH / "images" / "input_img.mha"
+        self.log(f"Copying data files to internal SPIDER data structure: {in_data_path} -> {internal_img_path}", level="NOTICE")
+        shutil.copy(str(in_data_path), str(internal_img_path))
+
+        filetype = str(in_data.type.ftype).lower()
+        modality = in_data.type.meta["mod"].upper()
+        is_modality_mr = modality == "MR"
+
+        if modality not in ["CT", "MR"]:
```

Comment:
This makes the code hard to follow.
I suggest two methods, `process_mr` and `process_ct`, that handle the two cases or two set-up methods, e.g., `setup_mr` and `setup_ct`, that set up the algorithm accordingly. 

Then, the task method can be cleaned up and organized into three sections:
- check modality
- setup algorithm
- run algorithm

Reply:
The new structure looks great!